hail/python/hail/methods/statgen.py

```diff
@@ -1983,6 +1998,446 @@ def _linear_skat(group,
+                   null_max_iterations: int = 25,
+                   null_tolerance: float = 1e-6,
+                   accuracy: float = 1e-6,
+                   iterations: int = 10000):
+    r'''The logistic sequence kernel association test (SKAT).
+
+    Logistic SKAT tests if the phenotype, `y`, is significantly associated with the genotype,
+    `x`. For :math:`N` samples, in a group of :math:`M` variants, with :math:`K` covariates, the
+    model is given by:
+
+    .. math::
+
+        \begin{align*}
+        X &: R^{N \times K} \\
+        G &: \{0, 1, 2\}^{N \times M} \\
+        \\
+        \textrm{logit}(P(y=1)) &= \beta_0 X + \beta_1 G + \varepsilon \quad\quad \varepsilon \sim N(0, \sigma^2)
+        \end{align*}
+
+    The usual null hypothesis is :math:`\beta_1 = 0`. SKAT tests for an association, but does not
+    provide an effect size or other information about the association.
+
+    Wu et al. argue that, under the null hypothesis, a particular value, :math:`Q`, is distributed
+    according to a generalized chi-squared distribution with parameters determined by the genotypes,
+    weights, and residual phenotypes. The SKAT p-value is the probability of drawing even larger
+    values of :math:`Q`. If :math:`\widehat{\beta_\textrm{null}}` is the best-fit beta under the
+    null model:
+
+    .. math::
+
+        \textrm{logit}(P(y=1)) = \beta_\textrm{null} X + \varepsilon \quad\quad \varepsilon \sim N(0, \sigma^2)
+
+    Then :math:`Q` is defined by Wu et al. as:
+
+    .. math::
+
+        \begin{align*}
+        r &= y - \textrm{logit}^{-1}(\widehat{\beta_\textrm{null}} X) \\
+        W_{ii} &= w_i \\
+        \\
+        Q &= r^T G W G^T r
+        \end{align*}
+
+    Therefore :math:`r`, the residual phenotype, is the portion of the phenotype unexplained by the
+    covariates alone. Also notice:
+
+    1. Each sample's phenotype is Bernoulli distributed with mean :math:`p_i` and variance
+       :math:`\sigma^2_i = p_i(1 - p_i)` where :math:`p_i` is the predicted probability under the
+       null model for sample :math:`i`.
+
+    2. :math:`G W G^T`, is a symmetric positive-definite matrix when the weights are non-negative.
+
+    We describe below our interpretation of the mathematics as described in the main body and
+    appendix of Wu, et al. According to the paper, the distribution of :math:`Q` is given by a
+    generalized chi-squared distribution whose weights are the eigenvalues of a symmetric matrix
+    which we call :math:`Z Z^T`:
+
+    .. math::
+
+        \begin{align*}
+        V_{ii} &= \sigma^2_i \\
+        W_{ii} &= w_i \quad\quad \textrm{the weight for variant } i \\
+        \\
+        P_0 &= V - V X (X^T V X)^{-1} X^T V \\
+        Z Z^T &= P_0^{1/2} G W G^T P_0^{1/2}
+        \end{align*}
+
+    Recall that the eigenvalues of a symmetric matrix and its transpose are the same, so we can
+    instead consider :math:`Z^T Z` (note that we elide transpositions of symmetric matrices):
```

Comment:
This isn't saying what you mean (`Z Z^T` is symmetric, so its transpose is `Z Z^T`). The only way I know to say this is that the eigenvalues of both `Z Z^T` and `Z^T Z` are the squared singular values of `Z`.